reflectivity_ui/interfaces/data_handling/data_manipulation.py

```diff
@@ -40,21 +40,34 @@ def generate_short_script(reduction_list):
             continue
 
         ws_name = "r%s" % reduction_list[i].cross_sections[xs].number
-        if len(api.mtd[ws_name]) == 0:
-            logging.info("  No entry in %s workspace group", ws_name)
+
+        # NOTE: It is possible that only one cross section is present in the run, therefore
+        #       api.mtd[ws_name] could be a Workspace2D instead of a workspace group.
+        #       Given that a modernization process is scheduled, we are going to do a ducktape
+        #       fixing here.
+        contain_single_crosssection = False
+        try:
+            if len(api.mtd[ws_name]) == 0:
+                logging.info("  No entry in %s workspace group", ws_name)
+        except:
```

Comment:
Bare exceptions are a bit dangerous. Change this to `except TypeError` and add a comment about assuming the object is actually a Workspace2D.

Reply:
The exception handling is adjusted as suggested.